util.go has a bug e should be minus one if len(d.Args[0]) == 0

when nginx conf like : "if ( $forward = "pre-release" ) {"
func prepareIfArgs(d Directive) Directive {
	e := len(d.Args) - 1
	if len(d.Args) > 0 && strings.HasPrefix(d.Args[0], "(") && strings.HasSuffix(d.Args[e], ")") {
		d.Args[0] = strings.TrimLeftFunc(strings.TrimPrefix(d.Args[0], "("), unicode.IsSpace)
		d.Args[e] = strings.TrimRightFunc(strings.TrimSuffix(d.Args[e], ")"), unicode.IsSpace)
		if len(d.Args[0]) == 0 {  // **here d.Args need to minus one** 
			d.Args = d.Args[1:]
		}
		if len(d.Args[e]) == 0 {
			d.Args = d.Args[:e]
		}
	}
	return d
}
 or else. program will panic and like this:
panic: runtime error: index out of range [4] with length 4

goroutine 1 [running]:
github.com/aluttik/go-crossplane.prepareIfArgs(0xc0000bcdee, 0x2, 0x168, 0xc0001e6090, 0x4, 0x7, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/lixiang/go/pkg/mod/github.com/aluttik/go-crossplane@v0.0.0-20201125234241-e4532485989f/util.go:60 +0x88b
github.com/aluttik/go-crossplane.(*parser).parse(0xc000071b08, 0xc0001000a0, 0xc0001020c0, 0xc0001cc3a0, 0x2, 0x2, 0x0, 0x2, 0x0, 0x0, ...)

--- a/parse_test.go
+++ b/parse_test.go
@@ -454,6 +454,69 @@ var parseFixtures = []parseFixture{
 			},
 		},
 	}},
+	parseFixture{"simple-with-if", "-ignore-directives-1", ParseOptions{IgnoreDirectives: []string{"listen", "server_name"}}, Payload{
+		Status: "ok",
+		Errors: []PayloadError{},
+		Config: []Config{
+			Config{
+				File:   filepath.Join("testdata", "simple-with-if", "nginx.conf"),
+				Status: "ok",
+				Errors: []ConfigError{},
+				Parsed: []Directive{
+					Directive{
+						Directive: "events",
+						Args:      []string{},
+						Line:      1,
+						Block: &[]Directive{
+							Directive{
+								Directive: "worker_connections",
+								Args:      []string{"1024"},
+								Line:      2,
+							},
+						},
+					},
+					Directive{
+						Directive: "http",
+						Args:      []string{},
+						Line:      5,
+						Block: &[]Directive{
+							Directive{
+								Directive: "server",
+								Args:      []string{},
+								Line:      6,
+								Block: &[]Directive{
+									Directive{
+										Directive: "location",
+										Args:      []string{"/"},
+										Line:      10,
+										Block: &[]Directive{
+											Directive{
+												Directive: "if",
+												Args:      []string{"$scheme", "=", "http"},
+												Line:      11,
+												Block: &[]Directive{
+													Directive{
+														Directive: "return",
+														Args:      []string{"200", "foo bar"},
+														Line:      12,
+													},
+												},
+											},
+											Directive{
+												Directive: "return",
+												Args:      []string{"200", "foo bar baz"},
+												Line:      14,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}},
 	parseFixture{"simple", "-ignore-directives-2", ParseOptions{IgnoreDirectives: []string{"events", "server"}}, Payload{
 		Status: "ok",
 		Errors: []PayloadError{},

--- a/testdata/simple-with-if/nginx.conf
+++ b/testdata/simple-with-if/nginx.conf
@@ -1,0 +1,17 @@
+events {
+    worker_connections 1024;
+}
+
+http {
+    server {
+        listen       127.0.0.1:8080;
+        server_name  default_server;
+
+        location / {
+            if ( $scheme = "http" ) {
+                return 200 "foo bar";
+            }
+            return 200 "foo bar baz";
+        }
+    }
+}

--- a/util.go
+++ b/util.go
@@ -50,6 +50,7 @@ func prepareIfArgs(d Directive) Directive {
 		d.Args[e] = strings.TrimRightFunc(strings.TrimSuffix(d.Args[e], ")"), unicode.IsSpace)
 		if len(d.Args[0]) == 0 {
 			d.Args = d.Args[1:]
+			e -= 1
 		}
 		if len(d.Args[e]) == 0 {
 			d.Args = d.Args[:e]

